src/block.rs

```diff
@@ -65,6 +69,7 @@ impl BlockBuilder {
         }
     }
 
+    // TODO: this may need adjustment, it returns a num of bytes + num of elements.
```

Comment:
yeah I think we need to multiple `self.offsets.len()` by SIZEOF_U16. We can also just use this method when allocating `buf` in `encode`.